packages/twenty-front/src/modules/spreadsheet-import/steps/components/MatchColumnsStep/components/UnmatchColumn.tsx

```diff
@@ -70,42 +27,70 @@ type UnmatchColumnProps<T extends string> = {
   onSubChange: (val: T, index: number, option: string) => void;
 };
 
+const StyledContainer = styled.div`
+  position: relative;
+  width: 100%;
+`;
+
+const StyledTransitionContainer = styled.div<{
+  isExpanded: boolean;
+  height: number;
+}>`
+  max-height: ${({ isExpanded, height, theme }) =>
+    isExpanded ? `calc(${height}px + ${theme.spacing(4)})` : '0'};
+  overflow: hidden;
+  position: relative;
+  transition: max-height 0.4s ease;
+`;
+
+const StyledContentWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: ${({ theme }) => theme.spacing(3)};
+  margin-top: ${({ theme }) => theme.spacing(4)};
+`;
+
 export const UnmatchColumn = <T extends string>({
   columns,
   columnIndex,
   onSubChange,
 }: UnmatchColumnProps<T>) => {
   const { fields } = useSpreadsheetImportInternal<T>();
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [contentHeight, setContentHeight] = useState(0);
+  const contentRef = useRef<HTMLDivElement>(null);
 
   const column = columns[columnIndex];
   const isSelect = 'matchedOptions' in column;
 
+  useLayoutEffect(() => {
```

Comment:
what is the purpose of this layoutEffect exactly?

Looking at the code I understand it is only to set the TransitionContainer height. If yes, I think we should extract this to a re-usable component like ExpandableContainer and hide this logic in the component

Reply:
We could add this component to twenty-ui so it's part of the toolbox

Reply:
Yes. The useLayoutEffect is just there to calculate the container’s height after the DOM updates so the TransitionContainer can adjust its height correctly.

I’ve extracted this logic into a new component, ExpandableContainer.  Adding it to twenty-ui sounds like a great idea.